pkg/server/server.go

```diff
@@ -357,3 +362,34 @@ func isSymlink(config string) bool {
 	}
 	return false
 }
+
+func setMasterRoleLabel(ctx context.Context, sc *Context, config *Config) error {
+	for {
+		nodeName := os.Getenv("NODE_NAME")
+		nodeController := sc.Core.Core().V1().Node()
+		node, err := nodeController.Cache().Get(nodeName)
+		if err != nil {
+			logrus.Infof("Waiting for master node %s startup: %v", nodeName, err)
+			time.Sleep(1 * time.Second)
+			continue
+		}
+		if len(node.Labels) == 0 {
+			node.Labels = make(map[string]string)
+		}
+		if _, ok := node.Labels[MasterRoleLabelKey]; ok {
```

Comment:
check that `node.Labels[MasterRoleLabelKey] == "true"` to ensure the value is correct, not just set